### PR TITLE
Mqt311 connection methods

### PR DESCRIPTION
**Pull request by sfod**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/WebsocketConnect/README.md [hunk 1]
```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
 ```sh
 mvn -P latest-release compile exec:java -pl samples/WebsocketConnect -Dexec.mainClass=websocketconnect.WebsocketConnect -Dexec.args="--endpoint <endpoint> --signing_region <signing region>"
 ```
+
+## Alternate connection configuration methods supported by AWS IoT Core
+
+### MQTT over WebSockets with static AWS credentials
+
+With a help of a static credentials provider your application can use a fixed set of AWS credentials. For that, you need
```
Comment (sbSteveK) [inline samples/WebsocketConnect/README.md:56]: Trivial: "With the help of"

File: samples/WebsocketConnect/README.md [hunk 2]
```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
...
+to instantiate the `StaticCredentialsProviderBuilder` class and provide it with the AWS credentials. The following code
+snippet demonstrates how to set up an MQTT3 connection using static AWS credentials for SigV4-based authentication.
+
```
Comment (sbSteveK) [inline samples/WebsocketConnect/README.md:59]: Trivial: Should we simplify this snippet? Include only what's absolutely necessary to establish this connection and remove references to "cmdData". 

e.g. remove withCertificateAuthorityFromPath, withCleanSession, withProtocolOperationTimeouts, (does this need client ID?). Change all cmdData related things to <clientID> or <endpoint>.

File: samples/WebsocketConnect/README.md [hunk 3]
```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
...
+        MqttClientConnection connection = builder.build();
+        return connection;
+    }
+}
+```
+
+### MQTT over WebSockets with Custom Authorizer
+
+An MQTT3 direct connection can be made using a [Custom Authorizer](https://docs.aws.amazon.com/iot/latest/developerguide/custom-authentication.html).
+When making a connection to a Custom Authorizer, the MQTT3 client can optionally passing username, password, and/or token
```
Comment (sbSteveK) [inline samples/WebsocketConnect/README.md:92]: Trivial: "When making a connection using a Custom Authorizer"

File: samples/WebsocketConnect/README.md [hunk 4]
```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
...
+signature arguments based on the configuration of the Custom Authorizer on AWS IoT Core.
+
+You will need to setup your Custom Authorizer so that the lambda function returns a policy document to properly connect.
+See [this page](https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html) on the documentation for
+more details and example return results.
+
+If your Custom Authorizer does not use signing, you don't specify anything related to the token signature and can use
+the following code:
+
+```java
```
Comment (sbSteveK) [inline samples/WebsocketConnect/README.md:102]: Same input as above snippet. Simplify by removing anything that's optional/unnecessary to establish this connection. Remove all references to cmdData and replace with what the customer needs to input.

File: samples/WebsocketConnect/README.md [hunk 5]
```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
...
+    } catch (Exception ex) {
+        throw new RuntimeException("Failed to create MQTT311 connection", ex);
+    }
+}
+```
+
+To run the websocket connect with custom authorizer use the following command:
+
+```sh
+mvn compile exec:java -pl samples/WebsocketConnect -Dexec.mainClass=websocketconnect.WebsocketConnect -Dexec.args="\
```
Comment (sbSteveK) [inline samples/WebsocketConnect/README.md:134]: I don't think these command line arguments are setup for the basic WebsocketConnect sample. e.g. `custom_auth_username`, `custom_auth_authorizer_name`, `custom_auth_password`. The websocket sample currently loads cmdData using "CommandLineUtils.SampleCommandLineData cmdData = CommandLineUtils.getInputForIoTSample("WebsocketConnect", args);" which in the util doesn't setup those specific arguments. We'll need to either build a specific separate sample that the CommandLineUtils loads and sets these arguments for, or we provide snippets for directly setting these arguments without use of cmdData.

File: samples/WebsocketConnect/README.md [hunk 6]
```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
...
+    } catch (Exception ex) {
+        throw new RuntimeException("Failed to create MQTT311 connection", ex);
+    }
+}
+```
+
+To run the websocket connect with custom authorizer using signing use the following command:
+
+```sh
+mvn compile exec:java -pl samples/WebsocketConnect -Dexec.mainClass=websocketconnect.WebsocketConnect -Dexec.args="\
```
Comment (sbSteveK) [inline samples/WebsocketConnect/README.md:177]: Same as above I think.

File: samples/WebsocketConnect/README.md (hunk 2 above)
Comment (sfod) [reply samples/WebsocketConnect/README.md:59]: Regarding `cmdData`: the idea was to provide a code snippet that could be injected into the sample as is. But considering your another comment that we also need to parse additional options, I tend to agree with you on this.

File: samples/WebsocketConnect/README.md (hunk 5 above)
Comment (sfod) [reply samples/WebsocketConnect/README.md]: I removed `cmdData`, so "How to run" became redundant

**Review by xiazhvera — APPROVED** [review]

**Pull request merged**